packages/core/src/config/load.ts

```diff
@@ -62,11 +62,15 @@ async function addConfigMetadata({
 }
 
 export async function loadConfig(
-  configPath: string | undefined = findConfig(),
-  customExtends?: string[],
-  processRawConfig?: (rawConfig: RawConfig) => void | Promise<void>
+  options: {
+    configPath?: string;
+    customExtends?: string[];
+    processRawConfig?: (rawConfig: RawConfig) => void | Promise<void>;
+    argv?: ArgsType;
+  } = {}
 ): Promise<Config> {
-  const rawConfig = await getConfig(configPath);
+  const { configPath = findConfig(), customExtends, processRawConfig, argv } = options;
+  const rawConfig = { ...(await getConfig(configPath)), ...getConfigArgv(argv), argv };
```

Comment:
I'd prefer to be more explicit:

```suggestion
  const config = await getConfig(configPath);
  const rawConfig = { 
       ...config, 
       region: config.region ?? argv.region,
       files: config.files ?? argv.files, 
       argv 
   };
```

Then we don't need `getConfigArgv` at all.

Reply:
For me, it's more cleaner. If we have much more argv to config, it will be long list of this.

Reply:
Indeed. However, we have to support this list anyways [here in types](https://github.com/Redocly/redocly-cli/pull/842/files#diff-623912bbe54cdd04399ad03ecf4c603f03682cf4d64116197426943f96811cf9R181-R189), and it will be even longer :)